API.md

```diff
@@ -114,3 +115,32 @@ npmlock2nix.build {
   };
 }
 ```
+
+### Source derivation attributes
```

Comment:
This addition seems sound. I just worry a bit about supporting this as an interface. Users should/will experience breakage over time as this is a very broad interface that we can't by no means keep stable/guarantee.

Reply:
I get the appeal and I can see why it might be used. I just don't want to advertise it as the go-to solution. We should do most if not all things for the users instead of exposing these broad interfaces...

I'll give it a bit of though.